### PR TITLE
add low level IDR encoding demo

### PR DESCRIPTION
**Pull request by chmike**
The Information Data Representation (IDR) is a work in progress. This code benchmarks the low level encoding/decoding functions to write custom value encoding/decoding. 

The ditp/idr/low package supports full time range with time offset and even time zone abbreviation encoding/decoding. For the benchmark I use the same nsec time encoding as benc.

### CODE REVIEW

File: benchmarks.go [hunk 1]
```diff
@@ -454,6 +455,15 @@ var benchmarkCases = []BenchmarkCase{
 		UnsafeStringUnmarshal: true,
 		TimeSupport:           TSNoSupport,
 		APIKind:               AKManual,
+	}, {
+		Name: "idr",
+		URL:  "github.com/chmike/ditp",
+		New:  idr.NewIDRSerializer,
+
+		BufferReuseMarshal: true,
```
Comment (matheusd) [inline benchmarks.go:463]: Given recent discussions which introduced this, I'd suggest adding both with and without buffer reuse versions of the benchmark.

File: benchmarks.go [hunk 2]
```diff
@@ -454,6 +455,15 @@ var benchmarkCases = []BenchmarkCase{
...
+		TimeSupport:        TSFullTzOffset,
+		APIKind:            AKManual,
+		Notes:              []string{"low level IDR encoding demo with same nsec time encoding as benc"},
```
Comment (matheusd) [inline benchmarks.go:466]: Capitalize the note ("Low level...") and you can drop the "same nsec time encoding"

**Comment by matheusd** [comment]
> The ditp/idr/low package supports full time range with time offset and even time zone abbreviation encoding/decoding. 

If the low level package supports it natively (i.e. `low.PutTime...`) I'd suggest using that instead of nsec to demonstrate it

**Comment by deneonet** [comment]
For demonstration, you should also rerun the benchmarks as instructed in the README.

**Comment by chmike** [comment]
I have applied your change requests and your suggestion to use the full time with tz offset.

But I don't see the point with the time encoding because we end up comparing apples with oranges. It is obviously slower and bigger with the full time encoding.

**Comment by chmike** [comment]
I did run the benchmark. Do you want me to commit it as well ?

**Comment by deneonet** [comment]
Yes.

**Comment by chmike** [comment]
Thank you.

**Pull request merged**